### PR TITLE
Add Proto types for storing TPM Platform Attestation in Collected Data

### PR DESCRIPTION
**Pull request by strideynet**
Part of https://github.com/gravitational/teleport.e/issues/1421

### CODE REVIEW

File: api/proto/teleport/devicetrust/v1/devicetrust_service.proto [hunk 1]
```diff
@@ -438,36 +439,6 @@ message TPMEnrollChallengeResponse {
   TPMPlatformParameters platform_parameters = 2;
 }
 
-// Encapsulates the value of a PCR at a point at time.
```
Comment (strideynet) [inline api/proto/teleport/devicetrust/v1/devicetrust_service.proto:441]: Moved to `tpm.proto`

**Review by Tener — APPROVED** [review]

File: api/proto/teleport/devicetrust/v1/device_collected_data.proto [hunk 2]
```diff
@@ -81,4 +82,16 @@ message DeviceCollectedData {
...
+  // If during the collection of this device data, the device performed a TPM
+  // platform attestation (e.g during enrollment or authentication), then this
+  // field holds the record of this attestation. This allows the state of the
+  // device to be compared to historical state, and allows for the platform
+  // attestations to be revalidated at a later date.
+  //
+  // This field is not explicitly sent up by the client, and any DCD sent by a
+  // client including this field should be rejected. The server should inject
+  // this field once verifying that the submitted platform attestation during
+  // the enrollment or authentication.
```
Comment (codingllama) [inline api/proto/teleport/devicetrust/v1/device_collected_data.proto:95]: Seems fair to say it's a "system managed" field, semantics are similar to others like it.

```suggestion
  // the enrollment or authentication.
  //
  // System managed.
```

File: api/proto/teleport/legacy/types/device.proto [hunk 3]
```diff
@@ -95,6 +95,35 @@ message DeviceCollectedData {
   string reported_asset_tag = 11 [(gogoproto.jsontag) = "reported_asset_tag,omitempty"];
   string system_serial_number = 12 [(gogoproto.jsontag) = "system_serial_number,omitempty"];
   string base_board_serial_number = 13 [(gogoproto.jsontag) = "base_board_serial_number,omitempty"];
+  TPMPlatformAttestation tpm_platform_attestation = 14 [(gogoproto.jsontag) = "tpm_platform_attestation,omitempty"];
+}
+
+// TPMPCR is the resource representation of TPMPCR
```
Comment (codingllama) [inline api/proto/teleport/legacy/types/device.proto:101]:
```suggestion
// TPMPCR is the resource representation of teleport.devicetrust.v1.TPMPCR.
```

**Review by codingllama — APPROVED** [review]

File: api/proto/teleport/devicetrust/v1/device_collected_data.proto [hunk 4]
```diff
@@ -81,4 +82,16 @@ message DeviceCollectedData {
...
   string base_board_serial_number = 13;
+
+  // If during the collection of this device data, the device performed a TPM
+  // platform attestation (e.g during enrollment or authentication), then this
+  // field holds the record of this attestation. This allows the state of the
+  // device to be compared to historical state, and allows for the platform
+  // attestations to be revalidated at a later date.
+  //
+  // This field is not explicitly sent up by the client, and any DCD sent by a
+  // client including this field should be rejected. The server should inject
```
[the remark this answers is not in the document]
Comment (strideynet) [reply api/proto/teleport/devicetrust/v1/device_collected_data.proto:93]: I'm erring towards the safer option of outright rejecting the request. There's no circumstance I expect the client to ever send this field, and I can imagine the only time someone would send this field would be a bad actor trying to inject a DCD entry with falsified PCRs in order to evade historical comparison checks on the PCRs.

File: api/proto/teleport/devicetrust/v1/device_collected_data.proto (hunk 2 above)
Comment (strideynet) [reply api/proto/teleport/devicetrust/v1/device_collected_data.proto:95]: Yup, `system managed` seems nicer.

File: api/proto/teleport/legacy/types/device.proto (hunk 3 above)
Comment (strideynet) [reply api/proto/teleport/legacy/types/device.proto]: 627a5a8b17

File: api/proto/teleport/devicetrust/v1/device_collected_data.proto (hunk 2 above)
Comment (strideynet) [reply api/proto/teleport/devicetrust/v1/device_collected_data.proto:95]: 627a5a8b17

**Pull request merged**